src/pem/_core.py

```diff
@@ -40,6 +40,16 @@ def as_bytes(self):
         """
         return self._pem_bytes
 
+    def __eq__(self, other):
+        return (type(self) == type(other) and
```

Comment:
Hmmm. I think raising `NotImplemented` only makes sense if you have an entirely different object? ie. `Certificate(...) == RSAPrivateKey(...)` should probably not be an error, otherwise you're almost always going to need to do your own type checks.